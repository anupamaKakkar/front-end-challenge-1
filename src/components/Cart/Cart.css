.cartContainer {
    display: flex;
    flex-direction: row;
    padding: 70px 85px 10px 90px;
}

label {
    font-weight: bold;
}
.infoProduct{
    width: auto;
    height: auto;
}
.infoProduct>label{
    display: block;
    margin-bottom: 10px;
    padding-top: 10px;
}
.pImage {
    flex-basis: 30%;
}

.productInfo {
    flex-basis: 70%;
    margin-left:30px;
}

.cTAButtons {
    display: inline-block;
    border: 1px solid;
    border-radius: 20px;
    background-color: #005beb;
    color: #fff;
    font-size: 15px;
    height: 40px;
    justify-content: center;
    outline: none;
    padding: 0 26px;
    cursor: pointer;
    margin-top: 15px;
}

.divider {
    display: block;
    height: 1px;
    width: 100%;
    border-color: #dfdfdf;
    border-top-width: 0;
    margin-top: 10px;
}

.cartTitle {
    margin: 25px auto;
    font-size: 36px;
    font-weight: bold;
    text-align: center;
}
@media (max-width: 767px) {
    .cartContainer{
        display: block;
        padding: 0px;
    }

    .pImage>img{
        margin: auto;
        display: block;
    }
    .productInfo{
        margin-left:85px;
    }
    .divider{
        display: none;
    }
    .cTAButtons{
        width: 75px;
        float: none;
        padding: 0px;
        justify-content: none;
    }
     
}
@media (max-width: 320px) {
    .productInfo{
        margin-left: 58px;
    }
}