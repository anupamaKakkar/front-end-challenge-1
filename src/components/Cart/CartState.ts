export interface CartState {
    quant?: number;
    productDetails?: any;
}