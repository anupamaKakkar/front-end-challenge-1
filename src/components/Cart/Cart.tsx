import * as React from 'react';
import { Route } from 'react-router-dom';

import { CartProps } from './CartProps';
import { CartState } from './CartState';
import './Cart.css';

export default class Cart extends React.Component<CartProps, CartState> {
    product: any = JSON.parse(localStorage.getItem("productDetail") || '{}');

    constructor(props: CartProps) {
        super(props);
        this.quantityChange = this.quantityChange.bind(this);

        this.state = {
            productDetails: this.product,
            quant: 1,
        }
    }

    componentDidMount() {
        this.setState({ productDetails: this.product })
    }

    quantityChange(event: React.FormEvent<HTMLInputElement>) {
        const value: any = event.currentTarget.value;
        if (value > 0) {
            this.setState({ quant: value });
        }
    }

    render() {
        const imgUrl: any = localStorage.getItem("productImage");

        return (
            <div>
                <div className="cartTitle">
                    <span>My Cart</span>
                </div>
                <div className="cartContainer">

                    <div className="pImage">
                        <img src={imgUrl} height="200" />
                    </div>
                    <div className="productInfo">
                        {this.state.productDetails &&
                            <div className="infoProduct">
                                <label>{this.state.productDetails.ProductTitle}</label>
                                <label>{this.state.productDetails.ProductBrand}</label>
                                <label>Size: {this.state.productDetails.ProductSize}</label>
                                <label>{this.state.productDetails.ProductSex}</label>
                                <label>Quantity: <input type="number" min='1' max='5' value={this.state.quant} onChange={this.quantityChange} /> </label>
                                {this.state.quant &&
                                    <label>Rs.{this.state.productDetails.ProductPrice * this.state.quant}</label>
                                }
                            </div>
                        }
                        <hr className="divider" />
                        <div className="cTALinks">
                            <Route render={({ history }) => (
                                <button
                                    className="cTAButtons"
                                    onClick={() => {
                                        history.push('/')
                                    }}
                                >Edit</button>
                            )} />
                            <Route render={({ history }) => (
                                <button
                                    className="cTAButtons"
                                    onClick={() => {
                                        localStorage.clear();
                                        history.push('/')
                                    }}>Cancel</button>
                            )} />

                            <button className="cTAButtons" onClick={() => {
                                alert("Order Submitted")
                            }}>
                                Submit
                            </button>

                        </div>
                    </div>
                </div>
            </div>
        )
    }
}