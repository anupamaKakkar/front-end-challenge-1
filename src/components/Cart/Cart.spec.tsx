import * as React from 'react';
import { shallow } from 'enzyme';

import Cart from './Cart';

describe('ProductDetails ', () => {

    it('should render correctly', () => {
        const wrapper = shallow(<Cart />);
        expect(wrapper).toBeDefined();
        expect(wrapper).toMatchSnapshot();
    });

    it('should chnage product quantity on click', () => {
        const wrapper = shallow(<Cart />);
        const component = wrapper.instance() as Cart;
        wrapper.find('input').simulate('change', {currentTarget: {value: 3}});
        expect(component.state.quant).toBe(3);
    });
});