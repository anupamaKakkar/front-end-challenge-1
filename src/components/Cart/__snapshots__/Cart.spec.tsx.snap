// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetails  should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cart />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="cartTitle"
        >
          <span>
            My Cart
          </span>
        </div>,
        <div
          className="cartContainer"
        >
          <div
            className="pImage"
          >
            <img
              height="200"
              src={null}
            />
          </div>
          <div
            className="productInfo"
          >
            <div
              className="infoProduct"
            >
              <label />
              <label />
              <label />
              <label>
                Quantity: 
                <input
                  max="5"
                  min="1"
                  onChange={[Function]}
                  type="number"
                  value={1}
                />
                 
              </label>
              <label>
                Rs.
                NaN
              </label>
            </div>
            <hr
              className="divider"
            />
            <div
              className="cTALinks"
            >
              <Route
                render={[Function]}
              />
              <Route
                render={[Function]}
              />
              <button
                className="cTAButtons"
                onClick={[Function]}
              >
                Submit
              </button>
            </div>
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            My Cart
          </span>,
          "className": "cartTitle",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My Cart",
          },
          "ref": null,
          "rendered": "My Cart",
          "type": "span",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="pImage"
            >
              <img
                height="200"
                src={null}
              />
            </div>,
            <div
              className="productInfo"
            >
              <div
                className="infoProduct"
              >
                <label />
                <label />
                <label />
                <label>
                  Quantity: 
                  <input
                    max="5"
                    min="1"
                    onChange={[Function]}
                    type="number"
                    value={1}
                  />
                   
                </label>
                <label>
                  Rs.
                  NaN
                </label>
              </div>
              <hr
                className="divider"
              />
              <div
                className="cTALinks"
              >
                <Route
                  render={[Function]}
                />
                <Route
                  render={[Function]}
                />
                <button
                  className="cTAButtons"
                  onClick={[Function]}
                >
                  Submit
                </button>
              </div>
            </div>,
          ],
          "className": "cartContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                height="200"
                src={null}
              />,
              "className": "pImage",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "height": "200",
                "src": null,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="infoProduct"
                >
                  <label />
                  <label />
                  <label />
                  <label>
                    Quantity: 
                    <input
                      max="5"
                      min="1"
                      onChange={[Function]}
                      type="number"
                      value={1}
                    />
                     
                  </label>
                  <label>
                    Rs.
                    NaN
                  </label>
                </div>,
                <hr
                  className="divider"
                />,
                <div
                  className="cTALinks"
                >
                  <Route
                    render={[Function]}
                  />
                  <Route
                    render={[Function]}
                  />
                  <button
                    className="cTAButtons"
                    onClick={[Function]}
                  >
                    Submit
                  </button>
                </div>,
              ],
              "className": "productInfo",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label />,
                    <label />,
                    <label />,
                    <label>
                      Quantity: 
                      <input
                        max="5"
                        min="1"
                        onChange={[Function]}
                        type="number"
                        value={1}
                      />
                       
                    </label>,
                    <label>
                      Rs.
                      NaN
                    </label>,
                  ],
                  "className": "infoProduct",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Quantity: ",
                        <input
                          max="5"
                          min="1"
                          onChange={[Function]}
                          type="number"
                          value={1}
                        />,
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Quantity: ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "max": "5",
                          "min": "1",
                          "onChange": [Function],
                          "type": "number",
                          "value": 1,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      " ",
                    ],
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Rs.",
                        NaN,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Rs.",
                      NaN,
                    ],
                    "type": "label",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "divider",
                },
                "ref": null,
                "rendered": null,
                "type": "hr",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Route
                      render={[Function]}
                    />,
                    <Route
                      render={[Function]}
                    />,
                    <button
                      className="cTAButtons"
                      onClick={[Function]}
                    >
                      Submit
                    </button>,
                  ],
                  "className": "cTALinks",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "render": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "render": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Submit",
                      "className": "cTAButtons",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Submit",
                    "type": "button",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="cartTitle"
          >
            <span>
              My Cart
            </span>
          </div>,
          <div
            className="cartContainer"
          >
            <div
              className="pImage"
            >
              <img
                height="200"
                src={null}
              />
            </div>
            <div
              className="productInfo"
            >
              <div
                className="infoProduct"
              >
                <label />
                <label />
                <label />
                <label>
                  Quantity: 
                  <input
                    max="5"
                    min="1"
                    onChange={[Function]}
                    type="number"
                    value={1}
                  />
                   
                </label>
                <label>
                  Rs.
                  NaN
                </label>
              </div>
              <hr
                className="divider"
              />
              <div
                className="cTALinks"
              >
                <Route
                  render={[Function]}
                />
                <Route
                  render={[Function]}
                />
                <button
                  className="cTAButtons"
                  onClick={[Function]}
                >
                  Submit
                </button>
              </div>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              My Cart
            </span>,
            "className": "cartTitle",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "My Cart",
            },
            "ref": null,
            "rendered": "My Cart",
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="pImage"
              >
                <img
                  height="200"
                  src={null}
                />
              </div>,
              <div
                className="productInfo"
              >
                <div
                  className="infoProduct"
                >
                  <label />
                  <label />
                  <label />
                  <label>
                    Quantity: 
                    <input
                      max="5"
                      min="1"
                      onChange={[Function]}
                      type="number"
                      value={1}
                    />
                     
                  </label>
                  <label>
                    Rs.
                    NaN
                  </label>
                </div>
                <hr
                  className="divider"
                />
                <div
                  className="cTALinks"
                >
                  <Route
                    render={[Function]}
                  />
                  <Route
                    render={[Function]}
                  />
                  <button
                    className="cTAButtons"
                    onClick={[Function]}
                  >
                    Submit
                  </button>
                </div>
              </div>,
            ],
            "className": "cartContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  height="200"
                  src={null}
                />,
                "className": "pImage",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "height": "200",
                  "src": null,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="infoProduct"
                  >
                    <label />
                    <label />
                    <label />
                    <label>
                      Quantity: 
                      <input
                        max="5"
                        min="1"
                        onChange={[Function]}
                        type="number"
                        value={1}
                      />
                       
                    </label>
                    <label>
                      Rs.
                      NaN
                    </label>
                  </div>,
                  <hr
                    className="divider"
                  />,
                  <div
                    className="cTALinks"
                  >
                    <Route
                      render={[Function]}
                    />
                    <Route
                      render={[Function]}
                    />
                    <button
                      className="cTAButtons"
                      onClick={[Function]}
                    >
                      Submit
                    </button>
                  </div>,
                ],
                "className": "productInfo",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <label />,
                      <label />,
                      <label />,
                      <label>
                        Quantity: 
                        <input
                          max="5"
                          min="1"
                          onChange={[Function]}
                          type="number"
                          value={1}
                        />
                         
                      </label>,
                      <label>
                        Rs.
                        NaN
                      </label>,
                    ],
                    "className": "infoProduct",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "label",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "label",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "label",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "Quantity: ",
                          <input
                            max="5"
                            min="1"
                            onChange={[Function]}
                            type="number"
                            value={1}
                          />,
                          " ",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        "Quantity: ",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "max": "5",
                            "min": "1",
                            "onChange": [Function],
                            "type": "number",
                            "value": 1,
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "input",
                        },
                        " ",
                      ],
                      "type": "label",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "Rs.",
                          NaN,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        "Rs.",
                        NaN,
                      ],
                      "type": "label",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "divider",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <Route
                        render={[Function]}
                      />,
                      <Route
                        render={[Function]}
                      />,
                      <button
                        className="cTAButtons"
                        onClick={[Function]}
                      >
                        Submit
                      </button>,
                    ],
                    "className": "cTALinks",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "render": [Function],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "render": [Function],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Submit",
                        "className": "cTAButtons",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Submit",
                      "type": "button",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
